internal/locate/region_cache_test.go

```diff
@@ -2259,13 +2260,35 @@ func (s *testRegionRequestToSingleStoreSuite) TestRefreshCache() {
 	s.cache.insertRegionToCache(&Region{meta: &r2, store: unsafe.Pointer(st), ttl: nextTTLWithoutJitter(time.Now().Unix())}, true, true)
 
 	r, _ = s.cache.scanRegionsFromCache(s.bo, []byte{}, nil, 10)
-	s.Equal(len(r), 2)
+	s.Equal(len(r), 1)
```

Comment:
This is not related to this PR I think.

Since region cache doesn't remove the first intersected region(it scan intersected region by `AscendGreaterOrEqual`), the outdated region (-inf, inf) is still alive and the new inserted valid region [{1}, inf) is ignored because the first seen region (-inf, inf) contains all the required ranges... We finally got outdated region though we know some valid regions.

Can you add a TODO in this test?